src/main/java/org/cactoos/io/SafeInput.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.InputStream;
+import org.cactoos.Input;
+
+/**
+ * A decorator of {@link Input} that prevents {@link InputStream}
+ * to be closed by its performers.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @since 1.0.0
+ */
+public final class SafeInput implements Input {
+
+    /**
+     * Origin.
+     */
+    private final Input origin;
+
+    /**
+     * Ctor.
+     * @param stream Stream to preserve
+     */
+    public SafeInput(final InputStream stream) {
+        this(() -> stream);
```

Comment:
@baudoliver7 why introduce this constructor? I would expect that one can use `InputOf` if needed. Actually, building an `Input` based directly on an `InputStream` is not the most general type of implementation for inputs, because that makes them one-use.

Reply:
@victornoel I totally agree with you ! Let's remove this ctor. :)